### PR TITLE
Refactor code duplicated between ciao-down and launcher

### PR DESCRIPTION
**Pull request by markdryan**
Ciao-down contained a number of functions that were copied directly from launcher.    These functions have been moved to different packages:

1. The existing qemu package
2. The new deviceinfo package

Launcher and ciao-down have been modified to use the functions in these packages.

### CODE REVIEW

**Comment by markdryan** [comment]
@sameo I'm adding a new file with one function to the qemu package in this PR in an effort to reduce code duplication between ciao-launcher and ciao-down. Any objections?

File: ciao-launcher/qemu.go [hunk 1]
```diff
@@ -159,43 +158,13 @@ func (q *qemuV) imageInfo(imagePath string) (int, error) {
...
-	}()
-
-	err := os.MkdirAll(dataDirPath, 0755)
-	if err != nil {
-		glog.Errorf("Unable to create config drive directory %s", dataDirPath)
-		return err
-	}
-
 	if len(metaData) == 0 {
 		defaultMeta := fmt.Sprintf("{\n  \"uuid\": %q,\n  \"hostname\": %[1]q\n}\n", cfg.Instance)
```
Comment (rbradford) [inline ciao-launcher/qemu.go]: Learnt something new...didn't know about %[n].

File: qemu/image.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	metaDataPath := path.Join(dataDirPath, "meta_data.json")
+	userDataPath := path.Join(dataDirPath, "user_data")
+
+	defer func() {
+		_ = os.RemoveAll(configDrivePath)
+	}()
+
+	err := os.MkdirAll(dataDirPath, 0755)
+	if err != nil {
+		return fmt.Errorf("Unable to create config drive directory %s : %v",
```
Comment (rbradford) [inline qemu/image.go]: I know this is just moving code around. But possibly in another patch you could introduce errors.Wrapf for this and the similar ones below.

**Comment by sameo** [comment]
@markdryan @rbradford LGTM

**Comment by rbradford** [comment]
Passed all BAT tests:

rob@singlevm:~/src/go/src/github.com/01org/ciao/_release/bat$ test-cases ./...
Package     Test Case                        Time Taken Result 
base_bat    TestGetTenants                   0.41s      PASS   
base_bat    TestGetClusterStatus             0.38s      PASS   
base_bat    TestGetWorkloads                 0.34s      PASS   
base_bat    TestGetInstance                  22.27s     PASS   
base_bat    TestStartAllWorkloads            72.01s     PASS   
base_bat    TestGetCNCIs                     16.10s     PASS   
base_bat    TestGetAllInstances              8.11s      PASS   
base_bat    TestDeleteAllInstances           11.29s     PASS   
image_bat   TestAddShowDelete                3.43s      PASS   
image_bat   TestDeleteNonExisting            0.38s      PASS   
image_bat   TestImageList                    4.57s      PASS   
storage_bat TestCreateBlockDevice            0.65s      PASS   
storage_bat TestCreateSizedBlockDevice       0.19s      PASS   
storage_bat TestCopyBlockDevice              1.60s      PASS   
storage_bat TestBootFromVolume               8.64s      PASS   
storage_bat TestStoppedInstance              8.52s      PASS   
storage_bat TestDeleteInUse                  8.45s      PASS   
storage_bat TestDetachRootFS                 6.20s      PASS   
storage_bat TestAddListDelete                12.71s     PASS   
storage_bat TestDeleteInstanceWhileAttaching 5.33s      PASS   
storage_bat TestDeleteBeforeDetached         12.55s     PASS   
storage_bat TestAttachDetach                 11.94s     PASS   
storage_bat TestAttachToRunningContainer     11.47s     PASS   
storage_bat TestDoubleAttach                 18.95s     PASS   
storage_bat TestAttachToStoppedInstance      10.10s     PASS   
storage_bat TestCreateVolumeFromImage        3.36s      PASS   
storage_bat TestCreateVolumeFromVolume       3.09s      PASS

**Pull request merged**